Sync with main

jneight opened the pull request.

mdomke committed:
Update AT banks

mdomke committed:
Update DE banks

mdomke committed:
Update BE banks

mdomke committed:
Update FI banks

mdomke committed:
Update PL banks

mdomke committed:
Add is_valid property to IBAN and BIC objects

mdomke committed:
chore: Allow line length of 100 chars in docs

mdomke committed:
Merge pull request #31 from mdomke/feature/add-is-valid-property

Add is_valid property to IBAN and BIC objects

mdomke committed:
Adapt IBAN registry script to updated SWFIT site

mdomke committed:
Update IBAN spec for Sao Tome and Principe

mdomke committed:
Update bank registry AT

mdomke committed:
Update bank registry BE

mdomke committed:
Update bank registry PL

mdomke committed:
Fix bank registry for HR

mdomke committed:
Fix listing of supported countries for BIC derivation

mdomke committed:
Poland's IBAN spec only has a branch-code but no bank-code

mdomke committed:
Add generated banks for spain

mdomke committed:
Fix IBAN generation for countries with branch/sort code

mdomke committed:
Merge branch 'feature/add-generated-spanish-banks' into master

* feature/add-generated-spanish-banks:
  Add generated banks for spain

mdomke committed:
Migrate from Travis to GitHub actions

mdomke committed:
Merge branch 'feature/migrate-to-github-actions' into master

* feature/migrate-to-github-actions:
  Migrate from Travis to GitHub actions

mdomke committed:
chore: This is the more appropriate name for the workflow

mdomke committed:
Run all tests on push

mdomke committed:
chore: basepython selection might trouble GitHub CI

mdomke committed:
Upload artifcats on tags

mdomke committed:
chore: URL friendlier workflow names

mdomke committed:
chore: Exchange travis badge with GitHub Actions workflow

insensitiveclod committed:
Added scraper script for NL-bic list (#35)

* Added NL BICs to bank_registry

* NL BIC support
removed unused variables from scraper

* NL BIC support
Removed unneeded data-mongering

mdomke committed:
Add BIC from IBAN test-case for NL

mdomke committed:
Adapt docs to mention NL support for BICs

mdomke committed:
Run linter and tests on pull requests

mdomke committed:
Give $default-branch macro another shot

mdomke committed:
Update bank registry AT

mdomke committed:
Update bank registry SI after NOVA KBM/ABANKA merge

mdomke committed:
chore: $default-branch macro not working

michal-michalak committed:
Fixed encoding for syncing PL bank registry (#37)

ckoehn committed:
Validate registry BICs (#40)

* Fix incorrect BIC

* Add test to validate registry BICs

mdomke committed:
Update IBAN registry

mdomke committed:
Update PL bank registry and fix data source errors

mdomke committed:
Update DE bank registry

mdomke committed:
Update AT bank registry

mdomke committed:
Update BE bank registry

mdomke committed:
Update NL bank registry

mdomke committed:
Update CZ bank registry

mdomke committed:
Update SI bank registry

petrboros committed:
Added SK banks (#39)

* Added SK banks

* Added generator for SK bank registry

* fixed typo

* used the last version of generator to generate the registry (stripped whitespace)

mdomke committed:
Exclude invalid SK banks

thorin-schiffer committed:
Ignore idea files

thorin-schiffer committed:
Added custom exceptions

thorin-schiffer committed:
Replaced ValueError with custom exceptions

thorin-schiffer committed:
Added exception detail comparison ignoring to pytest in doctests

mdomke committed:
Restructure documentation and switch to furo theme

mdomke committed:
Move examples to documentation only

mdomke committed:
Add project information to README

mdomke committed:
chore: Fix line length in documentation

mdomke committed:
Add troubleshooting section

mdomke committed:
Compare check digits if quick check passes

Check digits might still be invalid even if the condition

numeric(IBAN) % mod 97 == 1

applies. So the above is a necessary but not a sufficient condition.

mdomke committed:
Add Read The Docs configuration file

mdomke committed:
chore: Install package dependencies when building docs

mdomke committed:
Merge pull request #45 from mdomke/bugfix/compare-check-digits

Compare check digits if quick check passes

thorin-schiffer committed:
Fixes from the review

thorin-schiffer committed:
Merge branch 'master' into feature/custom-exceptions

thorin-schiffer committed:
Fixes after pull

mdomke committed:
Merge pull request #44 from eviltnan/feature/custom-exceptions

Add custom exceptions

mdomke committed:
Adapt tests and documentation to dedicated exception classes

mdomke committed:
Add missing releases to changelog

mdomke committed:
Move to PEP 517/518 compliant build system

mdomke committed:
chore: Fix formatting

mdomke committed:
chore: Update instructions for registry update scripts

mdomke committed:
chore: Pin black dependency in tests

mdomke committed:
Drop support for Python 2

mdomke committed:
Use pip installation method for building the docs

mdomke committed:
Finalize changelog for release

mdomke committed:
Add type hints

mdomke committed:
chore: Revert to Python 3.6 compliant typing

mdomke committed:
chore: Dropped support for Python 3.6

mdomke committed:
Merge branch 'feature/add-type-hints'

* feature/add-type-hints:
  chore: Dropped support for Python 3.6
  chore: Revert to Python 3.6 compliant typing
  Add type hints

mdomke committed:
Updated bank registry for various countries

mdomke committed:
Update changelog for release 2021.04.0

mdomke committed:
chore: Fix version compare link for 2021.04.0

mdomke committed:
Add manual selection of Lithuanian banks

mdomke committed:
chore: Remove Python 2.7 as test-target from Makefile

mdomke committed:
Adapt CHANGELOG for 2021.05.0 release

mdomke committed:
Add support for checksums in BBANs

mdomke committed:
Updated bank registry for AT, CZ, NL and PL

mdomke committed:
chore: Fix linting errors in docs

mdomke committed:
chore: Add return type annotation

mdomke committed:
Merge pull request #50 from mdomke/feature/add-account-code-checksum-support

Add account code checksum support

mdomke committed:
chore: Remove superfluous tick

mhemeryck committed:
Implements BE BBAN checksum check

mdomke committed:
Merge pull request #51 from mhemeryck/be_checksum_check

Implements BE BBAN checksum check

mdomke committed:
chore: Add changelog for 2021.05.2 release

mdomke committed:
Automatically cut BBAN based on IBAN spec for checksum calculation

mdomke committed:
chore: Add compare link for 2021.05.2 release to changelog

mdomke committed:
Fix README.rst reference in setup.cfg

jmfederico committed:
Script for Swedish bank registry

jmfederico committed:
Generates Swedish bank registry

jmfederico committed:
Include Sweden in list of BIC.domestic_bank_codes

mdomke committed:
Merge pull request #52 from jmfederico/patch-1

Script for Swedish bank registry

mdomke committed:
Adapt changelog for 2021.06.0 release

jmfederico committed:
Add Sweden as available for BIC class usage

jmfederico committed:
Add py.typed - PEP-561

mdomke committed:
Merge pull request #54 from jmfederico/patch-1

Add py.typed - PEP-561

mdomke committed:
Enable typing support as of PEP-0561

mdomke committed:
Add compare link for 2021.06.1

mdomke committed:
Add doc8 as pre-commit-hook

mdomke committed:
Upgrade black formatting to 21.6b0

mdomke committed:
Merge pull request #53 from jmfederico/patch-1

Add Sweden as available for BIC class usage

mdomke committed:
chore: Update mypy to 0.910

mdomke committed:
chore: Upgrade isort pre-commit hook

howorkon committed:
update some bank_registry

howorkon committed:
add some bank_registry

mdomke committed:
Merge pull request #58 from howorkon/master

add some bank registry

mdomke committed:
Add IBAN spec for Sudan (SD)

mdomke committed:
Update various bank registries and scripts

mdomke committed:
Reject installation for Python versions older than 3.7

mdomke committed:
Fix IBAN-BIC mapping for austrian federal institutes

mdomke committed:
Use importlib for loading resources

mdomke committed:
chore: importlib.resources.files is only available in Python 3.9

mdomke committed:
chore: Ignore mypy error on conditional imports

mdomke committed:
Merge pull request #60 from mdomke/use-importlib-for-loading-resources

Use importlib for loading resources

mdomke committed:
Ensure Belgian BBAN checksums are always 2 digits long

mdomke committed:
Added 440 additional bank records for spain

mdomke committed:
chore: Manually curated values are superseeded by generated registry

AntoineDefacto committed:
Update manual_fr.json

Add Swan to the manual_fr.json registry
See : https://www.iban.com/search-bic for more information on the bank

mdomke committed:
Merge pull request #62 from AntoineDefacto/patch-1

Update manual_fr.json

gautinils committed:
Added manual_is.json

jeremi-streammind committed:
Creating French BBAN (RIB) checksum algorithm

mdomke committed:
Merge pull request #64 from gautinils/master

Added manual_is.json

brunovila committed:
n26 neobank spain code

mdomke committed:
Merge pull request #68 from brunovila/master

n26 neobank spain code

mildamade committed:
generate iban with branch_code in source

mildamade committed:
correct parsing length

mdomke committed:
Merge pull request #71 from dvorakmilda/master

add branch_code at Czech IBAN

mdomke committed:
Add overwrite for CZ and FR BBAN positions

mdomke committed:
Update domestic bank registries

mdomke committed:
Update CHANGELOG for 2022.02.0 release

mdomke committed:
chore: Fix release links in changelog

TheoSinnige committed:
Skip empty lines in Dutch bank excel file.

mdomke committed:
Merge pull request #74 from TheoSinnige/feature/fix-nl-generation

#73 Skip empty lines in Dutch bank excel file.

mdomke committed:
Add __len__ to IBAN and BIC

Closes #72

mdomke committed:
Upadte bank registries

mdomke committed:
Merge pull request #67 from sholan/master

Creating French BBAN (RIB) checksum algorithm

mdomke committed:
chore: Cleanup french domestic checksum implemenation

mdomke committed:
chore: Fix doc links in changelog

mdomke committed:
Update code quality tools

imad3v committed:
Specify encoding when reading registry files

This fixes an issue that occurs on platforms that don't use UTF-8 as default encoding

mdomke committed:
Merge pull request #75 from imad3v/patch-1

Specify encoding when reading registry files

mdomke committed:
Remove empty entry in NL bank registry

Closes #77

mdomke committed:
chore: Fix quotes

mdomke committed:
Add CHANGELOG for 22.4.0 release

mdomke committed:
Updated bank registries for AT, CZ, DE, ES, PL and SK

mdomke committed:
chore: Fix link in CHANGELOG

mdomke committed:
Update troubleshooting

jose-reveni committed:
Update banco popular

mdomke committed:
Fix Italian BBAN checksum calculation

Closes #78

mdomke committed:
Add more tests for generating Italian IBANs

mdomke committed:
Merge pull request #81 from reveni-io/master

Update banco popular

Commit:
Fixed bank_code position for Jordan.
Source
https://www.cbj.gov.jo/EchoBusV3.0/SystemAssets/PDFs/EN/03-Instructions%20for%20implementing%20International%20Bank%20Account%20Number_Edited.pdf

Commit:
According to the Register of European Account Numbers (ECBS TR 201V3.9) (page 79), the Italian IBAN covers San Marino and so it also has a similar check digit.

jose-reveni committed:
Allow getting bank names from IBAN (#1)


Co-authored-by: Gonzalo Martín <98920740+gonzalo-reveni@users.noreply.github.com>

mdomke committed:
Merge pull request #80 from mdomke/bugfix/78-fix-italian-bban-checksum-calculation

Fix Italian BBAN checksum calculation

mdomke committed:
Merge pull request #83 from fabienpe/jordan-positions-fix

Fixed bank_code position for Jordan.

mdomke committed:
Merge pull request #84 from fabienpe/san-marino-checksum

Add check sum for San Marino

mdomke committed:
Add CHANGELOG for 22.4.1 release

jose-reveni committed:
Remove cached_property

jose-reveni committed:
Fix docstrings

mdomke committed:
Merge pull request #85 from reveni-io/master

Allow getting bank names from IBAN

mdomke committed:
Add CHANGELOG for 22.4.2 release

Commit:
feat: add bank of ireland

mdomke committed:
Merge pull request #86 from dnnsthnnr/feat/add-bank-of-ireland-to-registry

feat: add bank of ireland to registry

adamchainz committed:
Update package URL's

adamchainz committed:
Test on Python 3.10

adamchainz committed:
Remove 'hacking' from tox envs

mdomke committed:
Merge pull request #87 from adamchainz/patch-1

samizaman committed:
added Saudi Arabia bic/swift codes to bank_registry

Commit:
removed system and editor specific ignores from my user's ignore file

mdomke committed:
Merge pull request #93 from samizaman/saudi-swift-codes

added Saudi Arabia bic/swift codes to bank_registry

mdomke committed:
Merge pull request #88 from adamchainz/python_3.10

Test on Python 3.10

mdomke committed:
Merge pull request #89 from adamchainz/remove_hacking

Remove 'hacking' from tox envs

Krystofee committed:
Manually added Bulgarian BIC codes

mdomke committed:
Merge pull request #94 from Krystofee/master

Lukasz87 committed:
add pyinstaller support (#92)

mdomke committed:
chore: Add CHANGELOG for 2022.06.0 release

Krystofee committed:
Added Hungarian BIC codes with generator (#96)

Krystofee committed:
Added Romanian BIC codes with generator (#97)

* Added Romanian bic codes

* Updated update-all.sh with ro

Co-authored-by: Martin Domke <mail@martindomke.net>

dennisxtria committed:
feat(app): added more IE banks (#95)

mdomke committed:
chore: Update CHANGELOG for 2022.06.1 release

Krystofee committed:
Fixed Hungary bank codes (#98)

dchoroz committed:
feat(app): added GR banks (#99)

dchoroz committed:
feat(app): added CY banks (#100)

mdomke committed:
Add more tests for deriving BICs from bank codes

mdomke committed:
chore: Update bank registries

mdomke committed:
chore: Adapt CHANGELOG for 2022.6.2 release

Draugelis committed:
Add script to generate LT bank code registry (#101)

mdomke committed:
chore: Fix linter errors in LT registry script

mdomke committed:
chore: Update GitHub actions and Python version

mdomke committed:
chore: Remove manually curted Lithunin banks

mdomke committed:
chore: Add CHANGELOG for 2022.06.2 release

Krystofee committed:
Fixed HU bank registry generator (#102)

proper bank identifier is bank code (which is first three digits of branch code). There can be more entries which the same BIC code and different bank code.

mdomke committed:
chore: Add CHANGELOG for 2022.07.0 release

mdomke committed:
Fix IBAN.bank_name for countris without bank code

mkopec87 committed:
Senegal iban validation (#107)

* * Format files with pre-commit hook

* * Add Senegal IBAN verification

pebosi committed:
Switching to pandas (#103)

* Update requirements.txt

* use pandas for processing

* use pandas for processing

* process via pandas

* use pandas

* use pandas

* skip less

* use pandas

* Use pandas

* formatting, pandas changes

* format

* rm unused reqs

* fill_na values with empty string

* skip less

* change encoding

* no header no skip

* rm print, add mapping

Co-authored-by: pboden <peter.boden@cyber.bka.de>

mdomke committed:
chore: Regenerate bank registry

mdomke committed:
chore: Switch back to 3 digit bank codes for HU

mdomke committed:
chore: Fix formatting

jmfederico committed:
Update swedish bank registry (#108)

* Update script to fetch data from PDF

* Add Dockerfile to generate updated Swedish list

* Update list of Swedish banks

shpigunov committed:
Feature: Add Ukrainian banks (#114)

* UA Bank Registry generation script + generated registry

* Update requirements to ensure running all generation scripts

* add UA generation script to all scripts

* Changes as per review

Commit:
feat(bic): added NL bank (#119)

Commit:
fix(bic): added new IT banks (#122)

taoisu committed:
Add BIC registries (#111)

* get bank registries from api

* format

* Update ee

* Remove the script

mdomke committed:
Remove invalid UE entries

Commit:
fix: Correct bank code for Viva Payment (#127)

Co-authored-by: Kostas Ziovas <kostas@DESKTOP-60SBT2T>

Commit:
fix(bic): added new IT banks (#129)

Commit:
fix(bic): added new PT banks (#123)

Commit:
fix: Correct bank codes for Cypriot banks (#126)

Co-authored-by: Kostas Ziovas <kostas@DESKTOP-60SBT2T>

mdomke committed:
Explicitly include subpackages

Commit:
Updated BIC codes for GR, CY & added new ones for IT (#132)

* fix(bic): updated bic for GR, CY

* [SAVERS-1221]Add missing institutions for IT, CY

Co-authored-by: Anastasis Bourazanis <a.bourazanis@withplum.com>
Co-authored-by: Anastasios Bourazanis <a.bourazanis@gmail.com>

stegayet committed:
chore(ci): add Python 3.11 in test matrix (#133)

mdomke committed:
Add changelog for 2023.02.0 release

mhemeryck committed:
Belgium BBAN check fix (#136)

* Belgium BBAN check fix

The Belgian BBAN check has an issue when the checksum digits are exactly
"97". This is because the BBAN calculation is done using the "account
number" mod 97; normally, this would result in a checksum digit of "00",
but the standard in that case uses "97" instead.

Updated the checksum calculation to handle this case, plus updated the
related unit test.

* Docs update BE BBAN validation

Indicate in docs that Belgian BBAN is also checked.

dennisxtria committed:
chore(bank_registry): added new banks for IT, fixed codes for CY, PT (#134)

mdomke committed:
chore: Update generated bank registries

ezet committed:
Add norwegian bank registry (#137)

Also added script to generate the registry

mdomke committed:
Update bank registry for Austria

mdomke committed:
Update bank registry for Belgium

mdomke committed:
Update bank registry for Hungary

mdomke committed:
Update bank registry for Germany

mdomke committed:
Update bank registry for Poland

mdomke committed:
Update bank registry for Netherlands

mdomke committed:
Update bank registry for Slovenia

mdomke committed:
Update bank registry for Ukraine

mdomke committed:
Add changelog for 2023.03.0 release

mdomke committed:
chore: Update build badge URL

mdomke committed:
chore: Update GitHub workflow actions

alvarogf97 commented:
Guay 100K, mientras no toque el manual_es que en su momento le hice una PR al tio pero no me lo acepto

alvarogf97 commented on schwifty/bank_registry/manual_es.json:
```diff
@@ -1,130 +1,26 @@
 [
   {
-    "primary": true,
```
Este fichero es el problema, aqui es donde nosotros metimos los bancos

jneight commented:
tiene conflictos, así que con calma la iré revisando, por ahora no le he visto que toque nada de eso

mdomke committed:
Set bic to "" to avoid TypeError for Ukrainian banks

Fixes #143

mdomke committed:
Fetch the new bank registry for Germany

mdomke committed:
Update all available bank registries

stegayet committed:
chore(python): add Python 3.12 in test matrix (#149)

lukaszett committed:
Add missing banks (#145)

* Add Starling Bank

* Add ICARD AD

This file is supposed to be in alnum order - it doesn't seem to be.

* Add bunq bank

Once again not sure about the ordering in this file

* add UBS switzerland

Krystofee committed:
Added Italian banks registry (#152)

* Added Italian banks registry

* Removed manual_it.json

* Fixed Italian banks registry

* Fixed bic codes

* Parsing short names from bank names

---------

Co-authored-by: Martin Domke <mail@martindomke.net>

mdomke committed:
chore: Fix linting issues after merge

Natim committed:
Add a new BIC `candidates_from_bank_code` (#154)

* Add a new BIC candidates_from_bank_code to return a list of availables BIC from a given bank_codes.

* Use only bank_code index.

mdomke committed:
chore: accumulate does what build_list intends to do
